src/libcharon/plugins/kernel_super/kernel_super_ipsec.c

```diff
@@ -3036,7 +3224,9 @@ static void write_terminator_tunnels_oper(struct pm_terminator_conn* const termi
 			struct tunnel_counter_key *new_key;
 			INIT(new_key,
 				 .terminator_id = term_id,
-				 .super_peer_info = super_peer_info->get_ref(super_peer_info),
+				 .tenant_id = super_peer_info->get_tenant_id(super_peer_info),
+				 .account_id = super_peer_info->get_account_id(super_peer_info),
+				 .vrf_name = super_peer_info->get_vrf_name(super_peer_info),
```

Comment:
All these strings are not copied in the key. Meaning if the IKE SA is deleted after the iteration but before `oper_db_writer_set_terminator_tunnel_counters()` is called, these strings will point to deallocated memory.
You need to copy this strings into the key (and don't forget to free them in `destroy_tunnel_counter_key()`)